File: 2016/nazionali/paletta/att/grader.pas
uses paletta;

var	
	fr, fw : text;

{ Declaring variables }
	N : Longint;
	V : array of Longint;
	numero_ribaltamenti : Int64;

{ iterators used in for loops }
	i0: Longint;

begin
{$ifdef EVAL}
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
{$else}
    fr := input;
    fw := output;
{$endif}
    reset(fr);
    rewrite(fw);

	{ Reading input }
	read(fr, N);
	Setlength(V, N);
	for i0 := 0 to N-1 do
	begin
		read(fr, V[i0]);
	end;

	{ Calling functions }
	numero_ribaltamenti := paletta_sort(N, V);

	{ Writing output }
	writeln(fw, numero_ribaltamenti);
	
	close(fr);
    close(fw);
end.

File: 2016/nazionali/paletta/att/grader.c
#include <stdio.h>
#include <assert.h>
#include <stdlib.h>

static FILE *fr, *fw;

// Declaring variables
static int N;
static int* V;
static long long int numero_ribaltamenti;

// Declaring functions
long long int paletta_sort(int N, int* V);

int main() {
	#ifdef EVAL
		fr = fopen("input.txt", "r");
		fw = fopen("output.txt", "w");
	#else
		fr = stdin;
		fw = stdout;
	#endif

	// Iterators used in for loops
	int i0;

	// Reading input
	fscanf(fr, "%d ", &N);
	V = (int*)malloc(N * sizeof(int));
	for (i0 = 0; i0 < N; i0++) {
		fscanf(fr, "%d ", &V[i0]);
	}

	// Calling functions
	numero_ribaltamenti = paletta_sort(N, V);

	// Writing output
	fprintf(fw, "%lld\n", numero_ribaltamenti);
	
	fclose(fr);
	fclose(fw);
	return 0;
}
